cmd/podman/containers/cp.go

```diff
@@ -386,6 +386,7 @@ func copyToContainer(container string, containerPath string, hostPath string) er
 
 		_, err = io.Copy(tmpFile, os.Stdin)
 		if err != nil {
+			_ = tmpFile.Close()
```

Comment:
would it better to just unconditionally close directly after io.Copy(), then we do not need two Close() calls?

Reply:
we would still need to report the error from `tmpFile.Close()` when `io.Copy()` succeeds, so it would be similar to the following:

```diff
diff --git a/cmd/podman/containers/cp.go b/cmd/podman/containers/cp.go
index faeb7bd96..4093be57f 100644
--- a/cmd/podman/containers/cp.go
+++ b/cmd/podman/containers/cp.go
@@ -385,11 +385,12 @@ func copyToContainer(container string, containerPath string, hostPath string) er
                defer os.Remove(tmpFile.Name())
 
                _, err = io.Copy(tmpFile, os.Stdin)
+               errTmpClose := tmpFile.Close()
                if err != nil {
                        return err
                }
-               if err = tmpFile.Close(); err != nil {
-                       return err
+               if errTmpClose != nil {
+                       return errTmpClose
                }
 
                if !archive.IsArchivePath(tmpFile.Name()) {
```

we avoid one Close but it looks weirder to follow

Reply:
well why do we need to check for close() errors? The only thing that could happen there is if we try to close it twice and I really do not think this is a problem here.

Reply:
anyway I am good with merging it like this at well, was just thinking we can simply it a bit

Reply:
The temporary file is still written to the file system, so it is file system dependent but at least in theory some write errors might be returned by the kernel only on the subsequent `close()`.